File: apzkr-pzpi-21-2-chan-vadym/Task3-WebClient/eco-me-chan-front-end/src/services/ConsumptionService.js
// src/services/ConsumptionService.js
import apiInstance from './ApiService';

export const getUserConsumptionHistory = async (userId) => {
    try {
        const response = await apiInstance.get(`/api/Consumption/user/${userId}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching user consumption history:', error);
        throw error;
    }
};

File: apzkr-pzpi-21-2-chan-vadym/Task4-MobileClient/EcoMeChan_Mobile/Services/ConsumptionService.cs
using EcoMeChan_Mobile.Models;
using Newtonsoft.Json;
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;

namespace EcoMeChan_Mobile.Services
{
    public class ConsumptionService
    {
        private readonly HttpClient _httpClient;
        private readonly string _apiBaseUrl;

        public ConsumptionService()
        {
            _httpClient = new HttpClient();
            _apiBaseUrl = "https://525f-149-88-110-59.ngrok-free.app";
        }

        public async Task<List<Consumption>> GetUserConsumptionHistory(int userId)
        {
            var response = await _httpClient.GetAsync($"{_apiBaseUrl}/api/Consumption/user/{userId}");
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                var history = JsonConvert.DeserializeObject<List<Consumption>>(responseContent);
                return history;
            }
            else
            {
                // Handle error
                return null;
            }
        }
    }
}